accepted/0000-workspaces-part2.md

```diff
@@ -0,0 +1,221 @@
+- During the many discussions around this RFC a few strong points for named arguments stand out:
+  - Allow us to keep `npm workspace` for other command usages
+  - npm users are more familiar with usage of named configs, example: `npm test --workspace=foo` is very similar to `npm test --prefix=./packages/foo` - if you think on it in terms of "what workspace am I in" again, a named config makes more sense in npm
+  - [Organized a poll](https://github.com/npm/rfcs/issues/160) in which named arguments were the leading option, here are the results for reference:
+  ```
+  1. (12 votes) 21%  -  Positional argument with one-char alias: `npm workspace foo test` or `npm w foo test`
+  2. (11 votes) 11%  -  Positional argument but shorter default name: `npm ws foo test` or `npm w foo test`
+  3. (13 votes) 13%  -  Special character (operator-like): `npm :foo test`
+  4. (20 votes) 35%  -  Named argument: `npm --workspace=foo test` or `npm -w=foo test`
+  ```
+
+Given the results of all this preliminar work, the preferred way to run a command in the context of a workspace is to use the named `--workspace` argument or its `-w` short alias, e.g:
+
+In a project with the following structure:
+```
+./
+├── package.json { "name": "root", "workspaces": ["packages/foo", "packages/bar"] }
+└── packages/
+    ├── foo/
+    │   └── package.json { "name": "foo", "version": "1.0.0" }
+    └── bar/
+        └── package.json { "name": "foo", "version": "1.0.0" }
+```
+
+You can run tests for the `foo` workspace from the root of your project, with the following syntax:
+
+```
+npm test --workspace=foo
+```
+
+
+#### Install dependency example:
+
+Add `tap` as a **dependency** of a **workspace** named `foo`:
+
+```sh
+npm install tap --workspace=foo
+```
+
+
+Note: **Globs** are not supported as a valid `--workspace` argument value, the alternative is to use `npm <cmd> --workspace=<alias>` in which the `alias` can be defined beforehand in your **workspace** configuration as detailed in the following up section.
+
+### 3. Grouping/Aliasing workspaces
+
+One proposed feature that was surfaced during our brainstorm sessions is the ability for users to create arbitrary groups of **workspaces** defined via **workspaces configuration**, e.g:
+
+Given a **npm workspace** setup with the following file structure:
+
+```
+/root
+├── core
+│   ├── foo
+│   ├── bar
+│   └── util
+└── plugins
+    ├── helpers
+    ├── lorem
+    └── ipsum
+```
+
+It's possible to define a **groups** property within your **workspaces configuration** that will allow defining named sets of **workspaces** that can be later on used to run commands, e.g:
+
+```
+{
+    "name": "workspace-example",
+    "version": "1.0.0",
+    "workspaces": {
+        "groups": {
+            "core": ["core/*"], // accepts globs
+            "plugins": ["lorem", "ipsum"], // also accepts workspaces names
```

Comment:
Does workspace name === subpackage name? There might be some ambiguity between directory structure and package names, so maybe, everywhere it's specified, locations should be relative paths (`./packages/loren`) and package names regular strings? This would go for CLI args as well as aliases. Either should be able to use globs.